ui/app/src/elements/components/profile/with-profile.ts

```diff
@@ -0,0 +1,135 @@
+import { Profile, ProfilesSignal } from '@holochain-open-dev/profiles';
+import { css, CSSResult, html, PropertyValueMap } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import {
+  NHComponent,
+  NHProfileCard,
+  NHProfileIdenticon,
+} from '@neighbourhoods/design-system-components';
+import { contextProvided } from '@lit-labs/context';
+import { AsyncReadable, deriveStore, get, StoreSubscriber } from '@holochain-open-dev/stores';
+import { MatrixStore } from '../../../matrix-store';
+import { matrixContext, weGroupContext } from '../../../context';
+import { AgentPubKeyB64, AppSignal, DnaHash, decodeHashFromBase64, encodeHashToBase64 } from '@holochain/client';
+import { NHProfilePrompt } from './nh-profile-prompt';
+
+@customElement('with-profile')
+export class WithProfile extends NHComponent {
+  @contextProvided({ context: matrixContext, subscribe: true })
+  @state()
+  _matrixStore!: MatrixStore;
+
+  @contextProvided({ context: weGroupContext })
+  @state()
+  weGroupId!: DnaHash;
+
+  @property()
+  forAgentHash!: AgentPubKeyB64;
+
+  @state()
+  forAgentProfile!: Profile;
+
+  @property()
+  component!: 'card' | 'prompt' | 'identicon';
+
+  @state()
+  refreshed: boolean = false;
+
+  _profilesStore = new StoreSubscriber(this, () =>
+    this._matrixStore?.profilesStore(this.weGroupId as DnaHash),
+  );
+
+  _selectedNeighbourhoodProfile = new StoreSubscriber(
+    this,
+    () =>
+      deriveStore(
+        this._matrixStore.profilesStore(this.weGroupId as DnaHash),
+        store => store!.myProfile,
+      ),
+    () => [this.refreshed],
+  );
+
+  _forAgentProfile = new StoreSubscriber(
+    this,
+    () => this._profilesStore.value?.profiles.get(this.forAgentHash ? decodeHashFromBase64(this.forAgentHash) : this._matrixStore.myAgentPubKey) as AsyncReadable<Profile>,
+    () => [this.forAgentHash],
+  );
+
+  firstUpdated() {
+      this._profilesStore.value?.client.client.on("signal", (signal: AppSignal) => {
```

Comment:
This is a bit arcane. Are there any convenience functions we might be able to provide so that Applet devs don't have to understand so much of the `ProfilesStore` library and zome internals in order to listen for updates?

Reply:
There are also a couple of issues with this event listener. I'll list them out so they can be checked off:

- [ ] The listener never gets unbound so will slowly consume memory. Should be wired up to release itself upon component disconnect.
    - Note: guarding against these kinds of memory leaks is particularly important in the creation of an application like the NH launcher (as opposed to a website), since under normal operations the application is likely to be kept running for much longer.
- [ ] There is no error handling. I'm not sure what the Holochain client does with error handling in these callbacks but it's entirely possible it ignores them. The internal logic of this method should be wrapped in a try/catch and handled by the component such that it can display any errors to the viewer. Suspect this could assist in debugging #70.

Reply:
@adaburrows if you have opinions, please weigh in. I think this should probably be something that is handled in `SensemakerService` and pushes updates through `SensemakerStore` observables?